### PR TITLE
TS-19-v2-24: Refactor ESLint errors in redux directory

### LINKED ISSUE #24: Refactor: ESLint errors in redux,router,sagas directories
Refactor redux,router,sagas folders

###### _Acceptance Criteria:_
 There are no eslint errors in selected folders

###### _Sub Tasks \ Bugs:_ 
- [x] Redux folder refactored
- [x] Router folder refactored
- [x] Sagas folder refactored

Estimate:  _2 days_

### PR DESCRIPTION
**Pull request by Yashchemskyi-Oleksandr**

### CODE REVIEW

File: frontend/src/redux/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { createStore, applyMiddleware, compose } from 'redux';
```
Comment (andrii-kuzmichov) [inline frontend/src/redux/index.js]: maybe move this code to main file?

File: frontend/src/redux/reducers/departments.js [hunk 2]
```diff
@@ -7,102 +7,99 @@ const initialState = {
...
+                departments,
+                disabledDepartments,
             });
-        case actionTypes.GET_DEPARTMENT_BY_ID:
-            let department = state.departments.filter(
+        }
+        case actionTypes.GET_DEPARTMENT_BY_ID: {
+            let getDepartment = state.departments.filter(
                 (department) => department.id === action.result,
             )[0];
```
Comment (andrii-kuzmichov) [inline frontend/src/redux/reducers/departments.js]: use `find` instead of `filter`

File: frontend/src/redux/reducers/lesson.js [hunk 3]
```diff
@@ -23,31 +23,33 @@ const reducer = (state = initialState, action) => {
...
+            const lessons = state.lessons.filter((lesson) => lesson.id !== action.result);
             return updateObject(state, {
-                lessons: state.lessons,
+                lessons,
             });
-        case actionTypes.SELECT_LESSON_CARD:
-            let lesson = state.lessons.filter((lesson) => lesson.id === action.result)[0];
+        }
+        case actionTypes.SELECT_LESSON_CARD: {
+            let lesson = state.lessons.filter((less) => less.id === action.result)[0];
```
Comment (andrii-kuzmichov) [inline frontend/src/redux/reducers/lesson.js]: as above

File: frontend/src/redux/reducers/lesson.js [hunk 4]
```diff
@@ -23,31 +23,33 @@ const reducer = (state = initialState, action) => {
...
-            state.lessons.forEach((lesson) => {
+        }
+        case actionTypes.UPDATE_LESSON_CARD: {
+            const updatedLessons = state.lessons.map((lesson) => {
                 if (lesson.id === action.result.id) {
-                    lesson = { ...lesson, ...action.result };
+                    return { ...lesson, ...action.result };
                 }
-                updatedLessons.push(lesson);
+                return lesson;
```
Comment (andrii-kuzmichov) [inline frontend/src/redux/reducers/lesson.js]: refactor

**Review by andrii-kuzmichov — APPROVED** [review]

**Review by andrii-kuzmichov — APPROVED** [review]

**Review by Akr1s — APPROVED** [review]

**Review by HiroFive — APPROVED** [review]

**Pull request merged**